data/src/test/java/com/linkedin/data/schema/TestPdlBuilder.java

```diff
@@ -36,32 +38,57 @@
   @DataProvider
   private static Object[][] propertiesMapProvider()
   {
-    DataMap properties1 = new DataMap();
-    properties1.put("empty", new DataMap());
-    DataMap properties2 = new DataMap();
-    properties2.put("validate", properties1);
+    DataMap emptyProperty = new DataMap();
+    emptyProperty.put("empty", new DataMap());
+    DataMap arrayValueProperty = new DataMap();
+    arrayValueProperty.put("array", new DataList(Arrays.asList(1, 2, 3)));
+    DataMap flattenProperty = new DataMap();
+    flattenProperty.put("flatten", arrayValueProperty);
+    DataMap multipleProp = new DataMap();
+    multipleProp.putAll(emptyProperty);
+    multipleProp.putAll(arrayValueProperty);
+    DataMap jsonValueProp = new DataMap();
+    jsonValueProp.put("nested", multipleProp);
     return new Object[][]
         {
             {
-              properties1,
-              "@empty = {}\n"
+              emptyProperty,
+              "@empty = { }\n",
+              "@empty={}"
             },
             {
-              properties2,
-              "@validate.empty = {}\n"
-            }
-        //TODO Add test case for multiple properties in a map level once iteration logic is fixed to be deterministic
+              arrayValueProperty,
+              "@`array` = [ 1, 2, 3 ]\n",
+              "@`array`=[1,2,3]"
+            },
+            {
+              flattenProperty,
+              "@flatten.`array` = [ 1, 2, 3 ]\n",
+              "@flatten.`array`=[1,2,3]"
+            },
+            /* TODO Add test case for multiple properties in a map level once iteration logic is fixed to be deterministic
```

Comment:
I'd suggest adding a map test case with only 1 property per level, along with a TODO saying to add more properties at each level.

Reply:
flatten array property map is that use case and we need it to test compact logic.
that is why added separate test for json case.